### DISCUSSION TITLE
Enhancement: Add option to upload apps/containers icons manually

### DISCUSSION THREAD

**Discussion by AuthorShin** [IceWhaleTech/CasaOS#230] [Ideas]
It's very sad that most of the apps doesn't have pre build icon and even for some you can not find anywhere to add with URL and some times you need to upload one from your PC but no option for it. 
Would highly appreciate if this feature will consider in the future.

**Comment by Eirikur** [comment]
My take is that icons should always be customizable via right-click. It sounds like some better generic app icons are called for. I was about to suggest  a round robin so that sequentially added apps don't get the same icon, but would it not be better to have code that would dig into the app and find an icon before falling back to a generic icon?

**Reply by AuthorShin** [reply]
That could be a very nice solution, I personally after the container is running open it's webpage and with "View Source Page" on right click menu of the browser find the `apple-touch-icon` (usually 180px) and then add it to the container setting in casaOS as the Icon address like this: `http://192.168.1.54:4546/icons/apple-touch-icon-180x180.png`

**Comment by Protektor-Desura** [comment]
You should be able to upload the icon if it doesn't have one.
